examples/test-client/src/utils.ts

```diff
@@ -1,3 +1,31 @@
+import type { ClientConnectionInfo } from "./hooks/useParcnetClient";
+
 export function cn(...classes: string[]): string {
   return classes.filter(Boolean).join(" ");
 }
+
+export const DEFAULT_CONNECTION_INFO: ClientConnectionInfo = {
+  url: process.env.CLIENT_URL ?? "https://staging-rob.zupass.org",
+  type: "iframe"
+};
+
+export function getConnectionInfo(): ClientConnectionInfo {
+  let connectionInfo = DEFAULT_CONNECTION_INFO;
+  const storedConnectionInfo = localStorage.getItem("clientConnectionInfo");
+  if (storedConnectionInfo) {
+    try {
+      const parsedConnectionInfo = JSON.parse(
+        storedConnectionInfo
+      ) as ClientConnectionInfo;
+      if (
+        parsedConnectionInfo.type === "iframe" &&
+        typeof parsedConnectionInfo.url === "string"
+      ) {
+        connectionInfo = parsedConnectionInfo;
+      }
+    } catch (_e) {
+      // JSON parsing failed
```

Comment:
`console.error`?

Reply:
I've added a `console.error` here. The error is mostly inconsequential though - if parsing fails, we fall back to the default, and if the user changes the configuration then we'll save a new valid value.